### ISSUE TITLE
Keybind Setup Prompt Changes

### ISSUE THREAD

**Issue by Mid-The-Modder** [Mid-The-Modder/Midbox-Source#43]
I like the idea of disabling keybinds and all, but it still lacks customizability. I am considering remaking the keybind prompt to where you can change the keybinds, not disable them.

I think I know how I'll have to approach this. First, I'd have to make a system for when you input a keybind into a slot, it sets that keybind function to the keybind that was inputted. Then, to build the prompt. Once that is done, I'd need to go to **SongEditor.ts** to reorganize keybinds and unset them from their current positions (these current positions would likely be the default position). Then I'll have to find some way to insert code into the "if" statements for each keybind function depending on what the prompt has selected. Finally, some translation work and perhaps a "Default" button that sets all the keybinds to their defaults. 

This will probably end up taking me a while, but once the system is set up to where I can just add a keybind function to the list and tie them to **SongEditor.ts**, it's sailing.

**Comment by Mid-The-Modder** [comment]
As a reference, here is the current version at the time of sending this message.
![image](https://github.com/Mid-The-Modder/Midbox-Source/assets/141974713/345099a1-a9c5-47b4-94f1-0650d2037a8b)

It will change many times for what's to come.

**Comment by Mid-The-Modder** [comment]
Thinking of how to prep the **SongEditor.ts** keybind code for this.

What I'm thinking of so far is something like this:
// Get the selected keybind from the prompt, turn it into a number, and use that number in the case.
keybindNumberUndo = // Unsure. Call the selected option for the "Undo" keybind function. 
keybindNumberCut = // Unsure. Call the selected option for the "Cut" keybind function. 
case keybindNumberUndo:
                if (canPlayNotes) break;
                this._doc.undo();
                event.preventDefault();
                break;
case keybindNumberCut:
                if (canPlayNotes) break;
                this._doc.selection.cut();
                event.preventDefault();
                break;
// Continue...

My idea is to find a way to turn the selected options in the prompt into umbers for use in these 'empty' cases which contain the functions for what keybinds do.

**Comment by Mid-The-Modder** [comment]
Update to previous comment:
Not the optimal way to approach it. It would be better to perhaps create an enum for "Keycodes" and use that for turning the keybinds into numbers. 
![image](https://github.com/Mid-The-Modder/Midbox-Source/assets/141974713/f9ada78b-2018-45a5-ad2d-b0d06b519c23)


Hoping that in the future this prompt turns out amazing.
